src/map/utils/charutils.cpp

```diff
@@ -347,12 +347,12 @@ namespace charutils
      *                                                                       *
      ************************************************************************/
 
-    CCharEntity* LoadChar(uint32 charId)
+    std::unique_ptr<CCharEntity> LoadChar(uint32 charId)
     {
         TracyZoneScoped;
 
-        CCharEntity* PChar = new CCharEntity();
-        PChar->id          = charId;
+        std::unique_ptr<CCharEntity> PChar = std::make_unique<CCharEntity>();
```

Comment:
To make the rest of this function easier to use, you can do something like this:
```cpp
auto  charEntity = std::make_unique<CCharEntity>();
auto* PChar =  charEntity.get();
```

Then the rest of the function operates exactly the same way it used to without continually having to use `PChar.get()`